KMB/make_KMB_info.py

```diff
@@ -274,26 +319,42 @@ def get_source(self):
         return '[{url} {link_text}]\n{template}'.format(
             url=self.source, link_text=txt, template=template)
 
-    # @todo: use kommunkod/sockenkod to go via wikidata - T164576
-    def get_depicted_place(self):
-        """Get a linked version of the depicted place."""
-        s_triggers = ('a', 'o', 'u', 'å', 'e', 'i', 'y', 'ä', 'ö', 's', 'x')
+    def get_depicted_place(self, mappings):
+        """
+        Get a linked version of the depicted place.
+
+        If no 'land' is given the image is assumed to depict Sweden.
+
+        :param mappings: the shared mapping object
+        :return: depicted_place as wikitext
+        """
         depicted_place = None
         if not self.land or self.land == 'se':
-            kommun_link = self.kommunName
-            # add s if kommun does not end in s or a vowel
-            if not any(self.kommunName.endswith(x) for x in s_triggers):
-                kommun_link += 's'
-            kommun_link += ' kommun'
-            depicted_place = '{{Country|1=SE}}, %s, [[:sv:%s|%s]]' % (
-                self.lan, kommun_link, self.kommunName)
-
-            if self.socken:
-                template = 'User:Lokal_Profil/nycklar/socknar'
-                depicted_place += ', [[:sv:{{safesubst:%s|%s}}|%s (%s)]]' % (
-                    template, self.socken, self.sockenName, self.landskap)
+            if 'Gotland' in (self.lan, self.landskap) and not self.kommun:
+                # since lan/landskap/kommun are equivalent in this case
+                self.kommun = '0980'  # Gotlands kommun
+
+            depicted_place = '{{Country|1=SE}}'
+            if self.kommun:
+                kommun_id = '{:04d}'.format(int(self.kommun))  # zero pad
+                self.wd['kommun'] = mappings['kommun'][kommun_id]
+                depicted_place += ', {{city|%s}}' % self.wd['kommun']
+
+                if self.socken:
+                    socken_id = '{:04d}'.format(int(self.socken))  # zero pad
+                    self.wd['socken'] = mappings['socken'][socken_id]
+                    depicted_place += ', {{city|%s}}' % self.wd['socken']
+            else:
+                if self.lan:
+                    depicted_place += ', %s' % self.lan
+                elif self.landskap:
+                    depicted_place += ', %s' % self.landskap
+                else:
+                    self.meta_cats.add(
+                        'needing categorisation (no municipality)')
         else:
             depicted_place = '{{Country|1=%s}}' % self.land.upper()
+            self.meta_cats.add('needing categorisation (not from Sweden)')
 
         return depicted_place
```

Comment:
So the downside of the `upload_batches` repo is the lack of tests... might be worth reconsidering in the long term. Anyhow:

* 16000300013328 (normal case): `{{Country|1=SE}}, {{city|Q499410}}, {{city|Q10553813}}`
* 16001000328844 (land=cz): ` {{Country|1=CZ}}`
* 16001000151664 (no kommun, gotland): `{{Country|1=SE}}, {{city|Q374794}}, {{city|Q10656760}}`
* 16001000313224 (no socken): `{{Country|1=SE}}, {{city|Q149539}}`